File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Commands/CreateOrderCommand.cs
using MediatR;
using MediatorDesignPattern.Example2.Models;

namespace MediatorDesignPattern.Example2.Commands;

/// <summary>
/// MediatR Command - Request to create a new order
/// IRequest<T> interface tells MediatR that this is a request
/// </summary>
public class CreateOrderCommand : IRequest<Order>
{
    public string CustomerName { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal Price { get; set; }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Commands/CreateOrderCommand.ts
import { RequestData } from 'mediatr-ts';
import { Order } from '../Models/Order';

export class CreateOrderCommand extends RequestData<Order> {
  customerName: string;
  productName: string;
  quantity: number;
  price: number;

  constructor(customerName: string, productName: string, quantity: number, price: number) {
    super();
    this.customerName = customerName;
    this.productName = productName;
    this.quantity = quantity;
    this.price = price;
  }
}
